modules/microvm/virtualization/microvm/guivm.nix

```diff
@@ -89,6 +89,35 @@ let
             services.xdghandlers.enable = true;
           };
 
+          # Adding below systemd services to save power by turning off display when system is suspended / lid close
+          systemd.services.display-suspend = {
+            enable = true;
+            description = "Display Suspend Service";
+            serviceConfig = {
+              Type = "oneshot";
+              ExecStart = ''
+                ${pkgs.wlopm}/bin/wlopm --off '*' 
+              '';
+              Environment = ''
+                WAYLAND_DISPLAY=/run/user/1000/wayland-0 
+              '';
+            };
+          };
+
+          systemd.services.display-resume = {
+            enable = true;
+            description = "Display Resume Service";
+            serviceConfig = {
+              Type = "oneshot";
+              ExecStart = ''
+                ${pkgs.wlopm}/bin/wlopm --on '*'  
+              '';
+              Environment = ''
+                WAYLAND_DISPLAY=/run/user/1000/wayland-0 
+              '';
+            };
+          };
+
```

Comment:
Is the display-suspend service executed before host suspends/freezes the `microvm@gui-vm.service`?

Reply:
Yes, display suspend is called before pre-sleep actions.